oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/config/annotation/web/configurers/OidcClientRegistrationTests.java

```diff
@@ -428,10 +429,11 @@ public void requestWhenClientRegistersWithCustomMetadataThenSavedToRegisteredCli
 		RegisteredClient registeredClient = this.registeredClientRepository.findByClientId(
 				clientRegistrationResponse.getClientId());
 
-		assertThat(registeredClient.getClientSettings().<String>getSetting("custom-metadata-name-1"))
+		assertThat(registeredClient.getClientSettings().<String>getSetting("custom-metadata-1"))
 				.isEqualTo("value-1");
-		assertThat(registeredClient.getClientSettings().<String>getSetting("custom-metadata-name-2"))
+		assertThat(registeredClient.getClientSettings().<String>getSetting("custom-metadata-2"))
 				.isEqualTo("value-2");
```

Comment:
We should also assert that `clientRegistrationResponse` includes the `custom-metadata-1` and `custom-metadata-1` claims in the response.